### PR TITLE
OEL-3058: Check strings before deletion.

### PR DESCRIPTION
**Pull request by drishu**
### Description

Websites other than ewpp may not have the strings, so when the local storage tries to fetch them the result will be null.

reported also in https://github.com/openeuropa/oe_corporate_blocks/issues/158

### Change log

- Fixed: oe_corporate_blocks_post_update_40010

### CODE REVIEW

File: oe_corporate_blocks.post_update.php [hunk 1]
```diff
@@ -338,11 +338,17 @@ function oe_corporate_blocks_post_update_40010(&$sandbox): void {
...
-    $storage->delete($string);
-    $string = $storage->findString(['source' => 'EU institutions and bodies']);
-    $storage->delete($string);
-    $string = $storage->findString(['source' => 'https://european-union.europa.eu/institutions-law-budget/institutions-and-bodies/search-all-eu-institutions-and-bodies_en']);
-    $storage->delete($string);
+    $strings = [
+      ['source' => 'Search all EU institutions and bodies'],
+      ['source' => 'EU institutions and bodies'],
+      ['source' => 'https://european-union.europa.eu/institutions-law-budget/institutions-and-bodies/search-all-eu-institutions-and-bodies_en'],
+    ];
```
Comment (brummbar) [inline oe_corporate_blocks.post_update.php]: The 'source' key can be dropped and we keep just the value.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: oe_corporate_blocks.post_update.php (hunk 1 above)
Comment (drishu) [reply oe_corporate_blocks.post_update.php]: Not if we use StringDatabaseStorage, the keys of the arrays must match fields in the db table (lid, source, context, version), and it must be an array of conditions.

Comment (brummbar) [reply oe_corporate_blocks.post_update.php]:
```suggestion
    $strings = [
      'Search all EU institutions and bodies',
      'EU institutions and bodies',
      'https://european-union.europa.eu/institutions-law-budget/institutions-and-bodies/search-all-eu-institutions-and-bodies_en',
    ];
```
and then
```
$string = $storage->findString(['source' => $string]);

Comment (brummbar) [reply oe_corporate_blocks.post_update.php]: Let's also take the chance to rename $strings to $sources.

Comment (drishu) [reply oe_corporate_blocks.post_update.php]: clear enough, I misunderstood the first time

**Review by brummbar — APPROVED** [review]

**Review by nagyad — APPROVED** [review]

**Pull request merged**